File: Исходники/Глава 3. Часть 1/2. Тернарный поиск/C++/Олимпиадный/ternary search.cpp
double ternary_search(double f(double), double l, double r, double e)
{
	while (r - l > e)
		double a = (l * 2 + r) / 3
        double b = (l + r * 2) / 3
        if f(a) < f(b)
            r = b
        else
            l = a
    return (l + r) / 2
}

File: Исходники/Глава 3. Часть 1/2. Тернарный поиск/Python/ternary search.py
def ternarySearch(f, left: float, right: float, eps: float):
	while right - left > eps:
		a = (left * 2 + right) / 3
        b = (left + right * 2) / 3
        if f(a) < f(b)
            right = b
        else
            left = a
    return (left + right) / 2